inspirehep/modules/orcid/converter.py

```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""
+Handle conversion from INSPIRE records to ORCID.
+"""
+
+from __future__ import absolute_import, division, print_function
+from inspire_utils.record import get_value
+from inspire_utils.date import PartialDate
+from .builder import OrcidBuilder
+from flask import current_app
+import re
+from inspirehep.modules.hal.utils import (
+    get_journal_title,
+    get_publication_date,
+    get_conference_title,
+    get_conference_record,
+    get_conference_country,
+    get_doi,
+)
+
+
+class OrcidConverter(object):
+    """Coverter for the Orcid format."""
+
+    # Maps INSPIRE author roles to ORCID contributor roles
+    # for full list see: https://git.io/vdKXv#L235-L245
+    INSPIRE_TO_ORCID_ROLES_MAP = {
+        'author': 'author',
+        'editor': 'editor',
+        'supervisor': 'support-staff'  # TODO: Is this correct?
```

Comment:
no it's not. Since they don't have a role for supervisor, we should strip them (i.e. the authors who are supervisors) in the conversion.